### ISSUE TITLE
cpanfile contains invalid version number for Data::Dumper

### ISSUE THREAD

**Issue by pjwelsh** [Tux/Data-Peek#2]
https://metacpan.org/pod/Data::Dumper is currently at version 2.183. The Data-Peek cpanfile contains a recommends for a non-existent version 2.184.
This is throwing off the ability to automatically build Data-Peek into an RPM via the standard tools.

**Comment by Tux** [comment]
```
corelist -a Data::Dumper | grep -P '5\.[23][02468]'
  v5.20.0    2.151
  v5.20.1    2.151
  v5.20.2    2.151_01
  v5.20.3    2.151_01
  v5.22.0    2.158
  v5.22.1    2.158
  v5.22.2    2.158
  v5.22.3    2.158
  v5.22.4    2.158
  v5.24.0    2.160
  v5.24.1    2.160
  v5.24.2    2.160
  v5.24.3    2.160
  v5.24.4    2.160
  v5.26.0    2.167
  v5.26.1    2.167
  v5.26.2    2.167
  v5.26.3    2.167
  v5.28.0    2.170
  v5.28.1    2.170
  v5.28.2    2.170
  v5.28.3    2.170
  v5.30.0    2.174
  v5.30.1    2.174
  v5.30.2    2.174
  v5.30.3    2.174
  v5.32.0    2.174
  v5.32.1    2.174_01
  v5.34.0    2.179
  v5.34.1    2.179
  v5.36.0    2.184
  v5.36.1    2.184
  v5.38.0    2.188
```
```
$ grep Data::Dumper cpanfile
requires   "Data::Dumper";
recommends "Data::Dumper"             => "2.188";
```
The fact that the CPAN version of Data::Dumper is nut up-to-date is more something for the maintainer of the CPAN module. I /could/ try to check every update on CPAN, but that is not my responsibility I think.
So, your claim that version 2.184 is non-existent is wrong. The claim that it is not available on CPAN is quite correct and maybe worrysome. I will ping Nick, but I cannot guarantee that he will fix that.

**Comment by pjwelsh** [comment]
Do you actually have a dependency for a Data::Dumper version that seems to only be available to you?
That can't make for a very useful update for many :(
Please reconsider the version selection.
corelist -a Data::Dumper |tail -3
  v5.33.6    2.176     
  v5.33.7    2.176     
  v5.33.8    2.177

**Comment by Tux** [comment]
If you install 5.38.0 you will also have Data::Dumper 2.188. perl-5.38.0 is available to all
The output you paste means you do not have the most recent corelist, I am sure Module::CoreList is up-to-date op CPAN
```
$ perl -MV=Module::CoreList
Module::CoreList
        /pro/lib/perl5/site_perl/5.38.0/Module/CoreList.pm: 5.20230820
        /pro/lib/perl5/5.38.0/Module/CoreList.pm: 5.20230520
```

**Comment by pjwelsh** [comment]
Does  Data::Peek actually have a dependency on Data:Dumper 2.188 (and therefore perl v5.38)?
Sadly, my hands are tied as to the OS and perl version with the distro.
It does seem clear there may be a disconnect with CPAN.
I am not a perl-guy, but a systems person trying to make sure all of the extraneous/non-repository perl modules are at the latest versions possible as needed by the devs.
It is cool you are providing the command tools to help me better navigate the perl world. Thank you for the additional learning opportunity. Your effort in understanding this situation is much appreciated.

**Comment by Tux** [comment]
No it does not! `cpanfile` declares *any* version of Data::Dumper, but recommends the newest version I tested with.
If you have configured your system to always install recommended versions of modules, you might have a more serious issue.
`cpanfile` has three levels: `required`, `recommended`, and `suggested`. The `required is the minimum a module needs to function as documented, `recommended` is the level the module author would like to have (for whatever reason) and `suggested` is likely beyond what the module needs.
I myself use `required` for minimum version to work with (as it is intended for), `recommended` as the version that has no `CVE` issues and `suggested` as the most recent I tested with.

If you want Data::Peek installed and your configuration requires recommended versions - which is very much against CPAN policy if that would block installation - then just remove the recommends lines from `cpanfile`

**Comment by pjwelsh** [comment]
RPM packaged installs of the perl modules are required. The rpmbuild utility is seeing the value entry "2.188" as a requirement and is not flexible in its' understanding it seems.
I can certainly edit the cpanfile.
This has been the second time I've come across what I thought was an invalid version requirement. The first was actually a invalid version requirement. This issue seems to be different.
I don't understand how CPAN does not have the 2.188 version, but the perl 5.38 does. My only un-educated guess is that it is now included in the base perl 5.38?
Like you said, a query to Nick may help.
Thank you again for helping to clear this up.
